frontend/src/views/ensemble/SingleNote.tsx

```diff
@@ -18,24 +27,50 @@ interface SingleNoteProps {
   pitch: number;
 }
 
+const now = Tone.now();
 const PITCH_VALUES = ["C5", "B4", "A4", "G4", "F4", "E4", "D4", "C4"];
+let instrumentSampler = fluteSampler;
 
 const SingleNote: FC<SingleNoteProps> = ({ beatNumber, pitch }) => {
   // note type - attack sustain rest
   const [currentNoteType, setCurrentNoteType] = useState(NoteType.REST);
 
   const globalBeatNumber = useRecoilValue(beatNumberAtom);
+  const currentInstrument = useRecoilValue(currentInstrumentAtom);
   const playNow = globalBeatNumber === beatNumber;
+
   useEffect(() => {
+    // Do nothing if music is not being played
     if (!playNow) return;
+
     // Trigger a music note if we are not a NoteType.REST
-    // console.log("Playing Note")
-    // Insert Josh ...
-    if (currentNoteType !== NoteType.REST) {
-      const synth = new Synth().toDestination();
-      synth.triggerAttackRelease(PITCH_VALUES[pitch], "8n");
+    if (currentNoteType === NoteType.ATTACK) {
+      // Assign correct instrument sounds to what player selected
+      switch (currentInstrument) {
+        case Instrument.FLUTE:
+          instrumentSampler = fluteSampler;
+          break;
+        case Instrument.ALTO_SAX:
+          instrumentSampler = altoSaxSampler;
+          break;
+        case Instrument.MARIMBA:
+          instrumentSampler = marimbaSampler;
+          break;
+        case Instrument.GUITAR:
+          instrumentSampler = guitarSampler;
+          break;
+        case Instrument.BASS:
+          instrumentSampler = bassSampler;
+          break;
+        case Instrument.TUBA:
+          instrumentSampler = tubaSampler;
+          break;
+      }
+      instrumentSampler.triggerAttack(PITCH_VALUES[pitch]);
+    } else if (currentNoteType === NoteType.REST) {
+      instrumentSampler.triggerRelease(now);
```

Comment:
The `now` variable is defined outside of the component definiton. This means that this value of `now` is the same across all notes and never changes. 

Is this intentional? If so, is "now" the right name for a variable that doesn't change?

Reply:
I was testing Tone.now() in different places to get the attack of the note to stop when the current note is a rest. It was not changing anything and had just left it in the last place I tested it in. May need to look further into it as it was used in some examples.